gordo_components/client/client.py

```
@@ -266,6 +275,7 @@ def make_date_ranges(
 
         # Chunk over the dataframe by batch_size
         prediction_dfs = list()
+        error_messages = []
```

I am sceptical to this. I would rather have the logging be ugly but reliable. What if e.g. the program crashes hard after a error message has been "written", but before it gets to the big print-them-all loop?

---

~I agree, but practically, we've seen that the client might only fail on a handful of batch jobs. Whether for IO errors or whatever else. Seems exceptionally annoying for immediate and fatal errors to occur when the majority would work. Would you be open to a flag of some sort to indicate it should fail hard?  `--allow-failures`?~

---

ah, I mis-understood. How's about `logger.error` immediately, and some red text at the very end, basically re-printing the failed fetches as a _summary_ or sorts?

---

Added that, lemme know what your opinion is.